nixos/testing: restrict arguments to makeTest

ncfavier opened the pull request:
Disallow passing arbitrary arguments to `makeTest` since they are not used; this can help catch mistakes.

Then fix those mistakes, as well as some unrelated errors and warnings.

ncfavier committed:
nixos/testing: restrict arguments to makeTest

Disallow passing arbitrary arguments to makeTest since they are not
used; this can help catch mistakes.

ncfavier committed:
nixos/tests: fix some evaluation errors

Fixes errors caught by "nixos/testing: restrict arguments to makeTest" as
well as some unrelated errors and warnings.

ncfavier commented on nixos/tests/without-nix.nix:
```diff
@@ -4,12 +4,6 @@ import ./make-test-python.nix ({ lib, ... }: {
     maintainers = [ ericson2314 ];
   };
 
-  nixpkgs.overlays = [
-    (self: super: {
-      nix = throw "don't want to use this";
-    })
-  ];
-
```
@Ericson2314 note that the test fails if I add this to the actual machine configuration.

roberth replied:
Got nerd-sniped and fixed it.

ncfavier committed:
fixup! nixos/testing: restrict arguments to makeTest

roberth committed:
nixos: Make config.nix.enable pass test

roberth commented:
@ofborg test without-nix installer.simple

ncfavier commented on nixos/tests/without-nix.nix:
```diff
@@ -4,14 +4,23 @@ import ./make-test-python.nix ({ lib, ... }: {
+        nix_2_5 = throw "don't want to use pkgs.nix_2_5";
+        nix_2_6 = throw "don't want to use pkgs.nix_2_6";
+        nixFlakes = throw "don't want to use pkgs.nixFlakes";
+        nixStable = throw "don't want to use pkgs.nixFlakes";
+        nixUnstable = throw "don't want to use pkgs.nixFlakes";
```
Mismatch intentional?

roberth replied:
Good catch! Thanks :)

roberth committed:
lib: applyIfFunction -> applyModuleArgsIfFunction

roberth committed:
lib: Add toFunction

roberth committed:
nixosTest: Fix invocation

toFunction isn't quite the same as callPackage, but the difference
should only be relevant for cross pkgs, which are generally not
testable anyway.

roberth committed:
tests.nixos-functions.nixosTest-test: Test callPackage-like behavior

roberth commented on pkgs/top-level/all-packages.nix:
```diff
@@ -33613,7 +33613,7 @@ with pkgs;
             else test;
           calledTest = if lib.isFunction loadedTest
-            then callPackage loadedTest {}
+            then loadedTest { inherit pkgs lib; }
```
This was supposed to be similar to applying the args in `import ./make-test-python.nix ({ pkgs, ...}: `.
I've added some commits to fix this and test it properly.

roberth committed:
Merge remote-tracking branch 'upstream/master' into tests-restrict-arguments

ncfavier commented:
LGTM!

roberth commented:
The chromium package wasn't buildable.

@ofborg test caddy ceph-multi-node cri-o

roberth commented:
Awesome. Thank you @ncfavier !
